### PR TITLE
Call `GapObj` instead of accessing `chi.values` (for group characters) or `f.map` (for group homomorphisms)

### PR DESCRIPTION
**Pull request by ThomasBreuer**
This is a follow-up of #3671.

### CODE REVIEW

File: src/Groups/group_characters.jl [hunk 1]
```diff
@@ -3045,8 +3047,8 @@ function symmetric_parts(characters::Vector{GAPGroupClassFunction}, n::Int)
     length(characters) == 0 && return eltype(typeof(characters))[]
     tbl = parent(characters[1])
     return [class_function(tbl, chi)
-            for chi in GAPWrap.SymmetricParts(GAPTable(tbl),
-                         GAP.GapObj([chi.values for chi in characters]), n)]
+            for chi in GAPWrap.SymmetricParts(GapObj(tbl),
+                         GapObj([GapObj(chi) for chi in characters]), n)]
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]: For this kind of conversion you could now also write this:
```suggestion
                         GapObj(characters; recursive=true), n)]
```

File: src/Groups/homomorphisms.jl [hunk 2]
```diff
@@ -118,12 +118,12 @@ If `check` is set to `false` then it is not checked whether the mapping
 defines a group homomorphism.
 """
 function hom(G::GAPGroup, H::GAPGroup, gensG::Vector, imgs::Vector; check::Bool = true)
-  vgens = GapObj([x.X for x in gensG])
-  vimgs = GapObj([x.X for x in imgs])
+  vgens = GapObj([GapObj(x) for x in gensG])
+  vimgs = GapObj([GapObj(x) for x in imgs])
```
Comment (fingolfin) [inline src/Groups/homomorphisms.jl]: Here's another place one could now just do a recursive conversion:
```suggestion
  vgens = GapObj(gensG; recursive=true)
  vimgs = GapObj(imgs; recursive=true)
```

**Review by fingolfin — APPROVED** [review]

File: src/Groups/group_characters.jl [hunk 3]
```diff
@@ -2182,14 +2188,14 @@ function induce(chi::GAPGroupClassFunction, tbl::GAPGroupCharacterTable)
 end
 
 function induce(chi::GAPGroupClassFunction, tbl::GAPGroupCharacterTable, fusion::Vector{Int})
-  ind = GAPWrap.InducedClassFunctionsByFusionMap(GAPTable(parent(chi)),
-          GAPTable(tbl), GapObj([chi.values]), GapObj(fusion))
+  ind = GAPWrap.InducedClassFunctionsByFusionMap(GapObj(parent(chi)),
+          GapObj(tbl), GapObj([GapObj(chi)]), GapObj(fusion))
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]:
```suggestion
          GapObj(tbl), GapObj([chi]; recursive=true), GapObj(fusion))
```

File: src/Groups/group_characters.jl [hunk 4]
```diff
@@ -2432,13 +2438,13 @@ coordinates(chi::GAPGroupClassFunction) = coordinates(QQFieldElem, chi)
 
 function coordinates(::Type{T}, chi::GAPGroupClassFunction) where T <: Union{Integer, ZZRingElem, QQFieldElem, QQAbElem}
     t = parent(chi)
-    GAPt = GAPTable(t)
+    GAPt = GapObj(t)
     if characteristic(t) == 0
       # use scalar products for an ordinary character
-      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([chi.values]))
+      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([GapObj(chi)]))
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]:
```suggestion
      c = GAPWrap.MatScalarProducts(GAPt, GAPWrap.Irr(GAPt), GapObj([chi]; recursive=true))
```

File: src/Groups/group_characters.jl [hunk 5]
```diff
@@ -2432,13 +2438,13 @@ coordinates(chi::GAPGroupClassFunction) = coordinates(QQFieldElem, chi)
...
     else
       # decompose a Brauer character
-      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([chi.values]), GapObj("nonnegative"))
+      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([GapObj(chi)]), GapObj("nonnegative"))
```
Comment (fingolfin) [inline src/Groups/group_characters.jl]:
```suggestion
      c = GAPWrap.Decomposition(GAPWrap.Irr(GAPt), GapObj([chi]; recursive=true), GapObj("nonnegative"))
```

**Review by fingolfin — APPROVED** [review]

**Comment by ThomasBreuer** [comment]
failure in `test (1.10, short, macOS, RPTU)`:
```
ERROR: LoadError: On worker 2:
Failed to precompile Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13] to "/Users/aaruni/.julia/compiled/v1.10/Oscar/jl_VGgVXt".
```

**Comment by lgoettgens** [comment]
> failure in `test (1.10, short, macOS, RPTU)`:
> 
> ```
> ERROR: LoadError: On worker 2:
> Failed to precompile Oscar [f1435218-dba5-11e9-1e4d-f1a5fab5fc13] to "/Users/aaruni/.julia/compiled/v1.10/Oscar/jl_VGgVXt".
> ```

This type of failure is currently discussed on slack. Restarting the job seemed to work in this particular case

**Pull request merged**